assets/js/src/core/modules/app/error-boundary/error-boundary.tsx

```diff
@@ -0,0 +1,69 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import React, { Component, type ErrorInfo, type ReactNode } from 'react'
+import { isEmpty } from 'lodash'
+import { Flex, Typography } from 'antd'
+
+interface IErrorBoundaryProps {
+  children: ReactNode
+  fallback?: ReactNode
+}
+
+interface IErrorBoundaryState {
+  hasError: boolean
+  error: Error | null
+}
+
+class ErrorBoundary extends Component<IErrorBoundaryProps, IErrorBoundaryState> {
+  constructor (props: IErrorBoundaryProps) {
+    super(props)
+
+    this.state = {
+      hasError: false,
+      error: null
+    }
+  }
+
+  static getDerivedStateFromError (error: Error): IErrorBoundaryState {
+    return { hasError: true, error }
+  }
+
+  componentDidCatch (error: Error, errorInfo: ErrorInfo): void {
+    console.log('Error caught by ErrorBoundary:', error, errorInfo)
+  }
+
+  render (): ReactNode {
+    const { children, fallback } = this.props
+    const { hasError, error } = this.state
+
+    if (hasError) {
+      if (!isEmpty(fallback)) return fallback
+
+      return (
+        <Flex
+          align='center'
+          gap={ 10 }
+          justify='center'
+          style={ { position: 'absolute', inset: 0 } }
+        >
+          <Typography>{error?.message ?? 'Something went wrong.'}</Typography>
```

Comment:
Yes for some cases a retry button would make sense, but I would not do this in the first version and also make it dependent on the error type.

For the additional error information I totally agree - with different types of errors we could also handle different levels of error details.

Reply:
@vin0401 @markus-moser 
I was thinking about the "try again/retry" button, whether in the context of ErrorBoundary or API error handling. In my opinion, it makes more sense for API errors (for example, to retry a request, which is particularly useful for handling network connection issues). I also agree with Markus that it would be better to add this functionality later, not in first version or when we have specific requirements for it.

Reply:
@vin0401 Agree with you. Think, we need to consider adding try/catch logic for them when we need to track it for some reason.